juju/bundle.py

```diff
@@ -436,7 +436,7 @@ async def run(self, context):
 
         entity_id = await context.charmstore.entityId(self.charm)
```

Comment:
I suspect the channel might need to be passed into charmstore.entityId too: https://github.com/juju/theblues/blob/a8cf043507cde71eba77727f2e6952d323349760/theblues/charmstore.py#L329